packages/chrome/src/utils/useFocusableMount.ts

```diff
@@ -0,0 +1,41 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import { useLayoutEffect, useRef, MutableRefObject } from 'react';
+import activeElement from 'dom-helpers/activeElement';
+
+export interface IUseFocusableMountProp {
+  isMounted?: boolean;
+  focusOnMount?: boolean;
+  restoreFocus?: boolean;
+  targetRef: MutableRefObject<HTMLElement | null>;
+}
+
+export function useFocusableMount({
+  isMounted,
+  focusOnMount,
+  restoreFocus,
+  targetRef
+}: IUseFocusableMountProp) {
+  const wasOpenRef = useRef<boolean>(false);
+  const triggerRef = useRef<HTMLElement | Element | null>();
+
+  useLayoutEffect(() => {
+    if (isMounted && focusOnMount && targetRef.current) {
+      triggerRef.current = activeElement();
+      wasOpenRef.current = true;
+      targetRef.current.focus();
+    }
+  }, [wasOpenRef, isMounted, focusOnMount, targetRef]);
+
+  useLayoutEffect(() => {
```

Comment:
Is there a particular reason why `useLayoutEffect` is used instead of `useEffect`?

Reply:
Initially `useLayoutEffect` was used to prevent this code block from running when server side rendering. [After re-reading the documention](https://reactjs.org/docs/hooks-reference.html#uselayouteffect), it looks like that is not the case, and is ran during SSR. Will update this to `useEffect` instead. Good call out!

Reply:
After doing some experimentation with this, I noticed flickering when the `Sheet` animation was running within the Chrome suite. Do you think focusing on the element would be better ran after the component did update? Any reason comes to mind for not using `useLayoutEffect`?

Reply:
From our conversation, we decided to stick to `useEffect`. Updated!